### PR TITLE
util: Convert Translate Timeline to TypeScript

### PR DESCRIPTION
**Pull request by panicstevenson**

### CODE REVIEW

**Comment by panicstevenson** [comment]
I accidentally opened this as a non-draft PR. Criticism extremely welcome.

File: util/translate_timeline.ts [hunk 1]
```diff
@@ -66,25 +71,34 @@ const run = async (args) => {
 
   // TODO: this block is very duplicated with a number of other scripts.
   const importPath = '../' + path.relative(process.cwd(), triggersFile).replace('.ts', '.js');
+  // TODO: Fix dynamic imports in TypeScript
+  // eslint-disable-next-line max-len
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
   const triggerSet = (await import(importPath)).default;
+  // eslint-disable-next-line max-len
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
   const replacements = triggerSet.timelineReplace;
```
Comment (quisquous) [inline util/translate_timeline.ts:80]:
```suggestion
  const replacements = (await import(importPath)).default?.timelineReplace;
```

File: util/translate_timeline.ts [hunk 2]
```diff
@@ -66,25 +71,34 @@ const run = async (args) => {
...
     lineToSync[event.lineNumber] = event;
 
   // Combine replaced lines with errors.
   const timelineLines = timelineText.split(/\n/);
-  for (let i = 0; i < timelineLines.length; ++i) {
-    const lineNumber = i + 1;
-    let line = timelineLines[i].trim();
+  let lineNumber = 0;
+  for (const timelineLine of timelineLines) {
+    lineNumber++;
```
Comment (quisquous) [inline util/translate_timeline.ts:100]:
```suggestion
  timelineLines.forEach((timelineLine, lineNumber) => {
```

File: util/translate_timeline.ts [hunk 3]
```diff
@@ -101,5 +115,5 @@ const run = async (args) => {
   }
 };
 
-const args = parser.parseArgs();
+const args = parser.parseArgs() as { locale: Lang; timeline: string };
```
Comment (quisquous) [inline util/translate_timeline.ts:118]: I guess there's no way around the `as` here.

Comment (panicstevenson) [reply util/translate_timeline.ts:113]: Not as far as I could tell

**Comment by quisquous** [comment]
oops sorry for the half suggestion

**Pull request merged**